invenio_communities/assets/semantic-ui/js/invenio_communities/members/invitations/InvitationResultItem.js

```diff
@@ -35,112 +30,122 @@ const RoleContent = ({ title, description }) => (
+    is_selected: initialValue === title,
     content: <RoleContent title={title} description={description} />,
   }));
 
   return (
-    <Dropdown
-      selection
+    <MemberDropdown
+      updateMember={onChange}
+      initialValue={initialValue}
       options={options}
-      value={currentValue}
-      onChange={onChange}
     />
   );
 };
 
-export class InvitationResultItem extends Component {
-  constructor(props) {
-    super(props);
-  }
+const formattedTime = (expires_at) =>
+  DateTime.fromISO(expires_at).setLocale(i18next.language).toRelative();
+
+export const InvitationResultItem = ({
+  invitation,
+  onRoleChange,
+  isLoading,
+  key,
+  error,
+  success,
+  onTimerFinish,
+  onErrorClose,
+  onView,
+  onReInvite,
+  onCancel,
+}) => {
+  const { receiver, status, expires_at, links } = invitation;
 
-  render() {
-    const {
-      invitation,
-      onRoleChange,
-      isLoading,
-      key,
-      error,
-      success,
-      currentRole,
-      onErrorClose,
-    } = this.props;
-    const { receiver, status, expires_at } = invitation;
+  const role = _sample(roles);
 
-    const formattedTime = DateTime.fromISO(expires_at)
-      .setLocale(i18next.language)
-      .toRelative();
+  const avatar = links.avatar || "/static/images/square-placeholder.png";
 
-    return (
-      <Table.Row className="community-member-item" key={key}>
-        <Table.Cell>
-          <Grid textAlign="left" verticalAlign="middle">
-            <Grid.Column width={4}>
-              <Image
-                src={receiver.links.avatar}
-                fallbackSrc="/static/images/square-placeholder.png"
-                avatar
-                circular
-              />
+  return (
+    <Table.Row className="community-member-item" key={key}>
+      <Table.Cell>
+        <Grid textAlign="left" verticalAlign="middle">
+          <Grid.Column width={4}>
+            <Image
+              src={avatar}
+              fallbackSrc="/static/images/square-placeholder.png"
+              avatar
+              circular
+            />
+          </Grid.Column>
+          <Grid.Column width={12}>
+            <List>
+              <List.Item as="a">{receiver.name || receiver.user}</List.Item>
+              <List.Item as="small">{receiver.description}</List.Item>
+            </List>
+          </Grid.Column>
+        </Grid>
+      </Table.Cell>
+      <Table.Cell>{status}</Table.Cell>
+      <Table.Cell>
+        <Grid>
+          <Grid.Row>
+            <Grid.Column width={13} verticalAlign="middle">
+              {formattedTime(expires_at)}
             </Grid.Column>
-            <Grid.Column width={12}>
-              <List>
-                <List.Item as="a">{receiver.name}</List.Item>
-                <List.Item as="small">{receiver.description}</List.Item>
-              </List>
+            <Grid.Column width={3} textAlign="right">
+              {isLoading && <Loader active inline="centered" size="small" />}
+              <Timer
```

Comment:
I am really confused on why is this component needed

Reply:
Answered above... at the same time, I am wondering if we can have something more understandable.
For example, instead of having a Timer cmp, have a `SuccessIcon` cmp which will disappear after some time? Maybe in `react-invenio-deposit` (or not)?

Reply:
we could move it to the react-invenio-forms as a common component, but only in case if it is used elsewhere, but I think it is not yet the case. I proposed some naming alternatives above

Reply:
i will refactor it into a more understandable component, we can discuss whether it should be moved to somewhere else